### PR TITLE
[JENKINS-72114] Fix serialization of `HudsonPrivateSecurityRealm.UserDetailsImpl`

### PR DESCRIPTION
**Pull request by Vlatombe**
See [JENKINS-72114](https://issues.jenkins.io/browse/JENKINS-72114).

### Proposed changelog entries

- Developer: `HudsonPrivateSecurityRealm` objects are now serializable.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java
Comment (jglick) [inline core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java:728]: I was about to suggest that for legibility the fields and ctor params and getters be reordered into a more natural sequence. I think the original was just following alphabetical order suggested by an IDE. But now I see that https://github.com/spring-projects/spring-security/blob/cc6010876f7ee9be748471f4987b2112160bed28/core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java#L45-L92 is using the same sequence!

**Review by jglick — APPROVED** [review]
FTR fixes a mistake in #4848.

**Review by Kevin-CB — APPROVED** [review]
At first, I was concerned about the `private final String password;` but it's only the encrypted password 
image

And for `authorities`, with a HudsonPrivateSecurityRealm, I only was able to get simple value like `anonymous` or `authenticated`

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
